src/Network/Email/AbstractTransport.php

```diff
@@ -1,3 +1,16 @@
 <?php
-// @deprecated Backward compatibility with 2.x, 3.0.x
-class_alias('Cake\Mailer\AbstractTransport', 'Cake\Network\Email\AbstractTransport');
+// @codingStandardsIgnoreFile
+namespace Cake\Network\Email;
+
+class_alias(\Cake\Mailer\AbstractTransport::class, AbstractTransport::class);
+
+if (class_exists(AbstractTransport::class)) {
+    return;
+}
```

Comment:
@thinkingmedia Probably, I should have shown you a more better example. I am not concerned about performance, but warnings and non-BC. See the following example:

```php
class A {}

class_alias('A', 'B');
var_dump(new A instanceof B);
return;

class B extends A {}
```

The above code is invalid. Because the class `B` has no interfaces and no traits, it is already available at line 3. So `class_alias()` call would raise a redeclaration warning. As a result, `new A instanceof B` would return `false`, because `B` is not an alias of `A`.

 It contrasts with the following:

```php
interface I {}
class A implements I {}

class_alias('A', 'B');
var_dump(new A instanceof B);

return;

class B extends A {}
```

In this example, class `B` is implementing an interface `I`. So it is not available at line 3 and `class_alias()` would define `B` as an alias of `A`. As a result, `new A instanceof B` would return `true`.

So strictly speaking, this change might break existing applications.